src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+
+    public static <F, S> Pair of(F first, S second) {
+        return new Pair<>(first, second);
+    }
+
+    F first;
+    S second;
+
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    public int hashCode() {
+        int prime = 31;
+        int result = 1;
+        result = prime + result * first.hashCode();
+        result = prime + result * second.hashCode();
+        return result;
+    }
+
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (this.getClass() != obj.getClass() || obj == null) {
```

Comment:
what will happen if obj is null?

Reply:
I will get false. I deleted this check

Reply:
Nope. You'll get NullPointerException as your getClass check goes first. put them in correct order as in 3.2 hw and I'll approve